lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -2192,6 +2253,10 @@ bool MemoryToRegisters::run() {
       if (!asi)
         continue;
 
+      StackList<SILValue> ownedValues(&f);
+      StackList<SILValue> guaranteedValues(&f);
+      collectStoredValues(asi, ownedValues, guaranteedValues);
```

Comment:
I think this warrants a brief explanation. Like...
Record stored values because promoting the store exposes a copy of the stored value. After promotion, OSSA canonicalization runs on these value to cleanup unnecessary copies.

Reply:
Done.